### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by oleksandrdziubiuk**
My solution

### CODE REVIEW

File: src/main/java/core/basesyntax/CreateFigure.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class CreateFigure {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/CreateFigure.java:5]:
```suggestion
public class FigureProducer {
```
classes are usually nouns not verbs

File: src/main/java/core/basesyntax/CreateFigure.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public static void getCreateFigure(Figure[] myFigureArray) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/CreateFigure.java:6]:
```suggestion
    public static void insertRandomFigure(Figure[] myFigureArray) {
```

File: src/main/java/core/basesyntax/CreateFigure.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        Random random = new Random();
+
+        int generateFigure = random.nextInt(5);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/CreateFigure.java:9]: 5 is a magic number, declare it as a constant

File: src/main/java/core/basesyntax/CreateFigure.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+        for (int i = 0; i < myFigureArray.length; i++) {
+
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/CreateFigure.java:12]: these empty lines are redundant, they make the code longer unnecessary

File: src/main/java/core/basesyntax/HelloWorld.java [hunk 5]
```diff
@@ -1,8 +1,21 @@
...
 public class HelloWorld {
+    public static void main(String[] args) {
+        Random random = new Random();
+
+        int arrayRandomLength = random.nextInt(9);
+
+        Figure[] myFigureArray = new Figure[arrayRandomLength];
+        CreateFigure.getCreateFigure(myFigureArray);
 
+        for (Figure figures : myFigureArray) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/HelloWorld.java:17]:
```suggestion
        for (Figure figure : myFigureArray) {
```

File: src/main/java/core/basesyntax/HelloWorld.java [hunk 6]
```diff
@@ -1,8 +1,21 @@
...
+import java.util.Random;
+
 /**
  * Feel free to remove this class and create your own.
  */
 public class HelloWorld {
+    public static void main(String[] args) {
+        Random random = new Random();
+
+        int arrayRandomLength = random.nextInt(9);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/HelloWorld.java:12]: better to make 9 constant as well

File: src/main/java/core/basesyntax/Triangle.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Override
+    public double getSquare() {
+        return side1 * side2 / 2.0;
+    }
+
+    @Override
+    public String getColor() {
+        return "red";
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Triangle.java:25]: don't hardcode this value, what if I want to have blue, or green figure, better to have it as property

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Comment by sanek98cv** [comment]
Fixed

File: src/main/java/core/basesyntax/ColorProducer.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorProducer {
+    public static final int RANDOM_COLOR = 10;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/ColorProducer.java:6]: why 10 if you have 7 colors?

File: src/main/java/core/basesyntax/Circle.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    @Override
+    public double getSquare() {
+        return Math.PI * radius * radius;
+    }
+
+    @Override
+    public Color getColor() {
+        return new ColorProducer().insertRandomColor();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Circle.java:23]: your figure has a color (it should be a field) just return its value in the method getColor()

File: src/main/java/core/basesyntax/ColorProducer.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+
+    public Color insertRandomColor() {
+        int randomGenerate = new Random().nextInt(RANDOM_COLOR);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/ColorProducer.java:9]:
```suggestion
        int colorIndex = new Random().nextInt(Color.values().length);
```
let's use enum length

File: src/main/java/core/basesyntax/Figure.java [hunk 11]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public interface Figure {
+    String drawFigure();
+
+    double getSquare();
+
+    Color getColor();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:8]: move this method to the abstract classs and create there a field color with setter and getter

File: src/main/java/core/basesyntax/FigureProducer.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureProducer {
+    public static final int MAGIC_NUMBER = 5;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureProducer.java:6]:
```suggestion
    private static final int FIGURES_QUANTITY = 5;
```
if we don't use variable anywhere else let's make private 
[magic number definition](https://help.semmle.com/wiki/display/JAVA/Magic+numbers%3A+use+defined+constant#:~:text=A%20magic%20number%20is%20a,for%20other%20programmers%20to%20understand.)

File: src/main/java/core/basesyntax/Triangle.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Override
+    public double getSquare() {
+        return side1 * side2 / 2.0;
+    }
+
+    @Override
+    public Color getColor() {
+        return new ColorProducer().insertRandomColor();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Triangle.java:25]:
```suggestion
        return color;
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Comment by sanek98cv** [comment]
Fixed

File: src/main/java/core/basesyntax/Circle.java [hunk 14]
```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class Circle extends Shape implements Figure {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Circle.java:3]:
```suggestion
public class Circle extends Shape {
```

File: src/main/java/core/basesyntax/Shape.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+public abstract class Shape {
+    private final Color color = new ColorProducer().insertRandomColor();
+
+    public Color getColorRandom() {
+        return color;
+    }
+
+    public abstract double getSquare();
+
+    public abstract Color getColor();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Shape.java:12]:
```suggestion
    public Color getColor() {
    return color;
    }
```

File: src/main/java/core/basesyntax/Shape.java [hunk 16]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Shape {
+    private final Color color = new ColorProducer().insertRandomColor();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Shape.java:4]:
```suggestion
    private final Color color;
```

File: src/main/java/core/basesyntax/Shape.java [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Shape.java:5]:
```suggestion
public Shape(Color color) {
  this.color = color;
}
```

File: src/main/java/core/basesyntax/Shape.java [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+    public Color getColorRandom() {
+        return color;
+    }
+
+    public abstract double getSquare();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Shape.java:10]: move this method to an interface

File: src/main/java/core/basesyntax/Shape.java [hunk 19]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Shape {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Shape.java:3]:
```suggestion
public abstract class Shape implements Figure {
```

File: src/main/java/core/basesyntax/Square.java [hunk 20]
```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class Square extends Shape implements Figure {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Square.java:3]:
```suggestion
public class Square extends Shape  {
```

File: src/main/java/core/basesyntax/Trapezoid.java [hunk 21]
```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class Trapezoid extends Shape implements Figure {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Trapezoid.java:3]:
```suggestion
public class Trapezoid extends Shape {
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Comment by sanek98cv** [comment]
Fixed

File: src/main/java/core/basesyntax/Shape.java (hunk 17 above)
Comment (sanek98cv) [reply src/main/java/core/basesyntax/Shape.java:5]: не зрозумів як реалізувати

Comment (sanek98cv) [reply src/main/java/core/basesyntax/Shape.java:5]: в подальшому, в класах наслідниках

File: src/main/java/core/basesyntax/Shape.java (hunk 16 above)
Comment (sanek98cv) [reply src/main/java/core/basesyntax/Shape.java]: поясни будь-ласка чому не можна відразу ініціалізувати?
дякую

File: src/main/java/core/basesyntax/Shape.java (hunk 17 above)
Comment (RomanDubovskyi) [reply src/main/java/core/basesyntax/Shape.java:5]: use super() keyword inside of the constructor in child classes to call parent constructor

**Review by RomanDubovskyi — APPROVED** [review]
fix comment about constructor, rest is good

**Pull request closed without merging**